app/src/components/Register.tsx

```diff
@@ -62,6 +68,45 @@ class Register extends React.Component<RegisterProps, RegisterState> {
     };
   }
 
+  componentDidMount (): void {
+    this.shouldComponentUpdate();
+  }
+
+  shouldComponentUpdate (): boolean {
+    if (this.state.artists) {
+      return false;
+    }
+
+    const Artists = this.props.drizzle.contracts.Artists;
+
+    Artists.methods.getArtistsTotal()
+      .call()
+      .then((total: number) => {
+        const artists = [];
+
+        let id = 0;
+
+        while (id < total) {
+          id++;
+
+          artists.push(Artists.methods.getArtist(id)
+            .call()
+            .then((info: string[]) => {
+              return {
+                name: info[0],
+                wallet: info[1],
+              };
+            }));
+        }
+
+        return Promise.all(artists);
+      })
+      .then((artists: Artist[]) => this.setState({ artists: artists }))
+      .catch((err: any) => console.log(err));
```

Comment:
Separate out into async function and then call that inside here?

Reply:
Also the same comment as below with the mapping trick.

Reply:
I might be missing the obvious but where do you want me to apply the mapping trick in this case? I don't really do any list to list transformations

Reply:
Turning from a stream of integers between 0 and total into an artist object { name, wallet }

Reply:
I would have to create a loop to create that list of integers between 0 to total. and then do a map. Not sure that would look cleaner